src/libs/browser-storage.js

```diff
@@ -31,12 +28,74 @@ export const setDynamic = (storage, key, value) => {
   }
 }
 
-export const removeDynamic = (storage, key) => {
+export const getStatic = (storage, key) => {
+  return maybeParseJSON(storage.getItem(key))
+}
+
+export const setStatic = (storage, key, value) => {
+  if (value === undefined) {
+    storage.removeItem(key)
+  } else {
+    forceSetItem(storage, key, JSON.stringify(value))
+  }
+}
+
+export const listenStatic = (storage, key, fn) => {
+  window.addEventListener('storage', e => {
+    if (e.storageArea === storage && e.key === key) {
+      fn(maybeParseJSON(e.newValue))
+    }
+  })
+}
+
+export const getDynamic = (storage, key) => {
+  const storageKey = `dynamic-storage/${key}`
+  const data = maybeParseJSON(storage.getItem(storageKey))
+  return data && data.value
+}
+
+export const setDynamic = (storage, key, value) => {
+  const storageKey = `dynamic-storage/${key}`
+  if (value === undefined) {
+    storage.removeItem(storageKey)
+  } else {
+    forceSetItem(storage, storageKey, JSON.stringify({ timestamp: Date.now(), value }))
+  }
+}
+
+export const listenDynamic = (storage, key, fn) => {
   const storageKey = `dynamic-storage/${key}`
-  storage.removeItem(storageKey)
+  window.addEventListener('storage', e => {
+    if (e.storageArea === storage && e.key === storageKey) {
+      const data = maybeParseJSON(e.newValue)
+      fn(data && data.value)
+    }
+  })
 }
 
-const withUserPrefix = key => `${getUser().id}/${key}`
+/**
+ * Returns a stateful object that manages the given storage location.
+ * Implements an atom-compaible interface, and can be passed to useAtom.
+ */
+
+export const staticStorageSlot = (storage, key) => {
```

Comment:
The multiplexing has to happen somewhere, and I don't think there's any reason to prefer doing it in our code as opposed to using the native listeners. Note that there's another layer of multiplexing happening here, since multiple components can connect (and disconnect) to a slot.